### PR TITLE
Issue #1316 msw model from imod5 data

### LINKED ISSUE #1316: Add ``imod.msw.MetaSwapModel.from_imod5_data``
Requirements
--------------- 

- Read existing ``para_sim.inp`` and set as simulation settings
- Set starttime and endtime based on imod5 data
- Initiate all packages from imod5 data
- Allow users to define their own settings dictionary, instead of forcing hardcoded settings.

### PR DESCRIPTION
**Pull request by JoerivanEngelen**
Fixes #1316

# Description
Adds ``MetaSwapModel.from_imod5_data``, currently only models can be written by turning off validation, that is still something to pick up.

- Adds ``MetaSwapModel.from_imod5_data``
- Converts ``Imod5DataDict`` TypeAlias to TypedDict, with which we can avoid a lot of type casting.
- Add masking and broadcasting utilities for MetaSWAP iMOD5 data. I could not easily make calls to the MODFLOW6 utilities for masking, as these assumed presence of a layer coordinate and had logic to deal with that, which is irrelevant for MetaSWAP.
- Adds ``MetaSwapActive`` dataclass to store masks
- Allow user-defined ``settings`` upon initialization, instead of only supporting default settings.
- Add ``validate`` argument to ``MetaSwapModel.write`` to turn off model validation upon writing.

# Checklist
- [x] Links to correct issue
- [x] Update changelog, if changes affect users
- [x] PR title starts with ``Issue #nr``, e.g. ``Issue #737``
- [x] Unit tests were added
- [ ] **If feature added**: Added/extended example

### CODE REVIEW

File: imod/msw/model.py [hunk 1]
```diff
@@ -282,3 +305,66 @@ def regrid_like(
...
+            Is also used to infer the starttime of the simulation.
+
+        Returns
+        -------
+        MetaSwapModel
+            MetaSWAP model imported from imod5 data.
+        """
+        extra_paths = imod5_data["extra"]["paths"]
+        path_to_parasim = find_in_file_list("para_sim.inp", extra_paths)
+        parasim_settings = read_para_sim(path_to_parasim)
```
Comment (Manangka) [inline imod/msw/model.py:339]: What if `path_to_parasim` is empty? Should this be put in an if statement?

File: imod/msw/utilities/parse.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+def _try_parsing_string_to_number(s: str) -> ScalarType:
+    """
+    Convert string to number:
+
+    "1" -> 1
+    "1.0" -> 1.0
+    "a" -> "a"
+    """
+    try:
```
Comment (Manangka) [inline imod/msw/utilities/parse.py:15]: I wouldn't nest try-except statements.
Im suprised there is not an exisiting python method which does this.

An alternative:
```
try:
  return int(s)
except ValueError:
  pass

try:
  return float(s)
except ValueError:
  pass

return s
```

File: imod/msw/utilities/parse.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            value = s
+    return value
+
+
+def read_para_sim(file: Path | str) -> dict[str, ScalarType]:
+    with open(file, "r") as f:
+        lines = f.readlines()
+        out = {}
+        for line in lines:
+            ll = line[0 : line.find("!")].split("=")
```
Comment (Manangka) [inline imod/msw/utilities/parse.py:30]: what does `ll` stand for? Could you give it a more descriptive name?

File: imod/tests/test_msw/test_infiltration.py [hunk 4]
```diff
@@ -283,11 +283,11 @@ def test_from_imod5_data(data_infiltration):
     imod5_data = {"cap": cap_data}
     actual_pkg = Infiltration.from_imod5_data(imod5_data)
 
-    ones_vars = ["bottom_resistance", "extra_storage_coefficient"]
-    expected = expected_pkg.dataset.drop_vars(ones_vars)
-    actual = actual_pkg.dataset.drop_vars(ones_vars)
+    hardcoded_vars = {"bottom_resistance": -9999.0, "extra_storage_coefficient": 1.0}
```
[the remark this answers is not in the document]
Comment (JoerivanEngelen) [reply imod/tests/test_msw/test_infiltration.py:286]: Good suggestion, I named it ``MaskValues`` in the end.

File: imod/msw/model.py (hunk 1 above)
Comment (JoerivanEngelen) [reply imod/msw/model.py:339]: All iMOD5 usecases contain this file in the "EXTRA" section in the projectfile: Without it they couldn't run their MetaSWAP model with iMOD5. If the para_sim.inp is missing an error will be thrown: "could not find para_sim.inp in list of paths: ..."

**Pull request merged**